var/spack/repos/builtin/packages/py-dask-ml/package.py

```diff
@@ -0,0 +1,86 @@
+
+    variant('docs', default=False, description='Build HTML documentation')
+    variant('xgboost', default=False, description='Deploys XGBoost alongside Dask')
+
+    extends('python')
+    depends_on('python@3.6:', type=('build', 'run'))
+
+    depends_on('py-setuptools', type='build')
+    depends_on('py-setuptools-scm', type='build')
+    depends_on('py-pip', type=('build', 'run'))
+
+    depends_on('py-dask+array+dataframe@2.4.0:', type=('build', 'run'))
+    depends_on('py-distributed@2.4.0:', type=('build', 'run'))
+    depends_on('py-numba', type=('build', 'run'))
+    depends_on('py-numpy@1.17.3:', type=('build', 'run'))
+    depends_on('py-pandas@0.24.2:', type=('build', 'run'))
+    depends_on('py-scikit-learn@0.23:', type=('build', 'run'))
+    depends_on('py-scipy', type=('build', 'run'))
+    depends_on('py-dask-glm@0.2.0:', type=('build', 'run'))
+    depends_on('py-multipledispatch@0.4.9:', type=('build', 'run'))
+    depends_on('py-packaging', type=('build', 'run'))
+
+    depends_on('py-black', type=('build', 'run'), when='+docs')
+    depends_on('py-coverage', type=('build', 'run'), when='+docs')
+    depends_on('py-flake8', type=('build', 'run'), when='+docs')
+    depends_on('py-graphviz', type=('build', 'run'), when='+docs')
+    depends_on('py-heapdict', type=('build', 'run'), when='+docs')
+    depends_on('py-ipykernel', type=('build', 'run'), when='+docs')
+    depends_on('py-ipython', type=('build', 'run'), when='+docs')
+    depends_on('py-isort', type=('build', 'run'), when='+docs')
+    depends_on('py-mypy', type=('build', 'run'), when='+docs')
+    depends_on('py-nbsphinx', type=('build', 'run'), when='+docs')
+    depends_on('py-nose', type=('build', 'run'), when='+docs')
+    depends_on('py-numpydoc', type=('build', 'run'), when='+docs')
+    depends_on('py-psutil', type=('build', 'run'), when='+docs')
+    depends_on('py-pytest', type=('build', 'run'), when='+docs')
+    depends_on('py-pytest-cov', type=('build', 'run'), when='+docs')
+    depends_on('py-pytest-mock', type=('build', 'run'), when='+docs')
+    depends_on('py-sortedcontainers', type=('build', 'run'), when='+docs')
+    depends_on('py-sparse', type=('build', 'run'), when='+docs')
+    depends_on('py-sphinx', type=('build', 'run'), when='+docs')
+    depends_on('py-sphinx-rtd-theme', type=('build', 'run'), when='+docs')
+    depends_on('py-sphinx-gallery', type=('build', 'run'), when='+docs')
+    depends_on('py-testpath', type=('build', 'run'), when='+docs')
+    depends_on('py-tornado', type=('build', 'run'), when='+docs')
+    depends_on('py-toolz', type=('build', 'run'), when='+docs')
+    depends_on('py-zict', type=('build', 'run'), when='+docs')
+    depends_on('py-dask-sphinx-theme@1.1.0:', type=('build', 'run'), when='+docs')
+    depends_on('py-nbsphinx', type=('build', 'run'), when='+docs')
+
+    depends_on('py-xgboost+dask', type=('build', 'run'), when='+docs')
```

Comment:
Why is this needed to build the docs?

Reply:
These seem to be needed to output `xgboost` related files. If not specified, some files will not be output.